c2corg_ui/static/js/gpxupload.js

```diff
@@ -68,7 +101,10 @@ app.GpxUploadController.prototype.importGpx_ = function(gpx) {
     featureProjection: 'EPSG:3857'
   });
   if (features.length > 0) {
-    this.scope_.$root.$emit('featuresUpload', features);
+    features = this.validatedFeatures_(features);
```

Comment:
What about moving this line above ``if (features.length > 0) {``?
This would avoid the second ``if (features) {``.

By the way are you sure ``if (features) {`` is enough? An empty array evaluates to ``true``:
```
> [] ? 'true': 'false'
"true"
```
I then recommend:
```
features = this.validatedFeatures_(features);
if (features.length) {
  this.scope_.$root.$emit('featuresUpload', features);
}
```